### DISCUSSION TITLE
network bandwidth usage

### DISCUSSION THREAD

**Discussion by NazarAli** [filamentphp/filament#8718] [Help]
i now use filament and livewire for learning and test 
not yet released application and i see the application used a lot of network bandwidth 
and i afraid my application slow in slow network .
can any one give me some knowledge

**Comment by ousid** [comment]
What do you mean, by your application slow in slow network?

- Did you write [unit-tests](https://coderflex.com/blog/all-you-need-to-know-to-start-using-phpunit-in-laravel-application) to make sure your app works correctly?
- Have you run a [code-coverage](https://php.watch/articles/php-code-coverage-comparison)
- Have you analyze your frontend, and the [load time](https://pagespeed.web.dev/) to see the page speed?
- Have you seen, and analyzed your bandwidth, and I/O statistics in the server?
- Have you checked your app against the known [n+1](https://laravel-news.com/laravel-n1-query-problems) problem
- Have you installed a debugger such as [Laravel debug bar](https://github.com/barryvdh/laravel-debugbar), or [Laravel Telescope](https://laravel.com/docs/10.x/telescope) to analyze your requests locally?
- Have you set up the permission correctly, when [deploying](https://coderflex.com/blog/2-ways-to-setup-file-permissions-for-laravel-in-production) to production?
 
A lot of questions, actually, you should ask yourself, when developing an app, or deploying to product. This is the basic ones, you need to start with them first.

**Reply by NazarAli** [reply]
no i mean it take a lot of  KB when load page or paginate
for example in paginate i show 500 records without any relation just has one field it take 4.2 MB 
and i disable all debugger and in production mode 

![Screenshot 2023-09-25 212149](https://github.com/filamentphp/filament/assets/7056487/738e07eb-5414-4d55-8669-83cb72b000fd)

**Reply by krekas** [reply]
This is how livewire full page components works. Also, I think you are seeing uncompressed size which isn't the size that browser gets.

**Reply by ousid** [reply]
Is this on a production server, or local?

**Reply by NazarAli** [reply]
in production mode

**Comment by NazarAli** [comment]
@krekas 
thank i user deflated_module in wamp it reduce the response from 4.2 MB to 76 KB is there another way to more reduce the response size ??

**Reply by krekas** [reply]
I don't know. Maybe some specific server settings for nginx or what you will be using

**Reply by NazarAli** [reply]
i use apache server
